src/main/java/dev/enjarai/trickster/screen/ScrollAndQuillScreenHandler.java

```diff
@@ -36,17 +37,21 @@ public class ScrollAndQuillScreenHandler extends ScreenHandler implements Revisi
     public final SyncedProperty<SpellPart> spell = createProperty(SpellPart.class, SpellPart.ENDEC, new SpellPart());
     public final SyncedProperty<SpellPart> otherHandSpell = createProperty(SpellPart.class, SpellPart.ENDEC, new SpellPart());
     public final SyncedProperty<Boolean> isMutable = createProperty(Boolean.class, true);
+    @SuppressWarnings("unchecked") // I hate Java generics, sorry about the horrid cast -- Aurora
+	public final SyncedProperty<Hamt<Pattern, SpellPart>> macros = (SyncedProperty<Hamt<Pattern, SpellPart>>) (Object) createProperty(Hamt.class, Endec.map(Pattern.ENDEC, SpellPart.ENDEC).xmap(Hamt::fromMap, Hamt::asMap), Hamt.empty());
```

Comment:
Why is this cast needed? :concern: Feels like there should be a better way...

Reply:
The `SyncedProperty` maker takes a `Class<T>` and those can't have generic info. Java generics, what can you do?

Reply:
It's ugly here, but I'd hope it at least makes its usages cleaner.